be/src/runtime/routine_load/data_consumer.cpp

```diff
@@ -444,6 +444,14 @@ Status PulsarDataConsumer::init(StreamLoadContext* ctx) {
         _custom_properties.emplace(item.first, item.second);
     }
 
+    std::string log_file_path = config::sys_log_dir + "/pulsar-cpp-client.log";
+    if (config::pulsar_client_log_level >= 0 && config::pulsar_client_log_level <= 3) {
+        config.setLogger(new pulsar::FileLoggerFactory(
```

Comment:
The memory of this newly created object pulsar::FileLoggerFactory leaks, am I right?

Reply:
According to the implementation of pulsar C++ client, this point will be maintained by:

```
struct ClientConfigurationImpl {
    ...
    std::unique_ptr<LoggerFactory> loggerFactory;
    ...
    std::unique_ptr<LoggerFactory> takeLogger() { return std::move(loggerFactory); }
};

ClientConfiguration& ClientConfiguration::setLogger(LoggerFactory* loggerFactory) {
    impl_->loggerFactory.reset(loggerFactory);
    return *this;
}
```